src/git-base.ts

```diff
@@ -0,0 +1,73 @@
+import {Command} from '@oclif/command'
+import {AxiosBitBucketClient, RepositoryReference} from './api/service/axios-bitbucket-client'
+import {AxiosJiraClient} from './api/service/axios-jira-client'
+import {AxiosFactory} from './api/service/axios-factory'
+import {SpawnOptions, spawn} from 'child_process'
+import * as winston from 'winston'
+import * as Config from '@oclif/config'
+
+export abstract class GitBaseCommand extends Command {
+  // eslint-disable-next-line no-useless-escape
+  static JIRA_ISSUE_KEY_REGEX = /(([^\/]+\/)+)?(?<issueKey>[^-]+-\d+)/gm;
+
+  jira?: AxiosJiraClient
+
+  bitBucket?: AxiosBitBucketClient
+
+  logger: winston.Logger
+
+  constructor(argv: string[], config: Config.IConfig) {
+    super(argv, config)
+    this.logger = winston.createLogger({
+      levels: winston.config.cli.levels,
+      format: winston.format.combine(
+        winston.format.splat(),
+        winston.format.printf(info => `${info.level}: ${info.message}`)
+      ),
+      transports: [new winston.transports.Console()],
+    })
+  }
+
+  async _spawn(command: string, argsv?: readonly string[], options?: SpawnOptions): Promise<{status: number; stdout: string; stderr: string}> {
```

Comment:
This is not related to a GitBaseCommand. This is probably better in some kind of utility class.

Reply:
separated out into `util/child-process.ts`